lib/grandpa/message.go

```diff
@@ -77,21 +77,55 @@ func (v *VoteMessage) ToConsensusMessage() (*ConsensusMessage, error) {
 	}, nil
 }
 
-// NeighbourMessage represents a network-level neighbour message
-type NeighbourMessage struct {
-	Version byte
-	Round   uint64
-	SetID   uint64
-	Number  uint32
+// VersionedNeighbourMessage represents the enum of nighbor messages
+type VersionedNeighbourMessage scale.VaryingDataType
+
+// Index Returns VDT index
+func (VersionedNeighbourMessage) Index() uint { return 2 }
+
+func newVersionedNighborMessage() VersionedNeighbourMessage {
+	vdt := scale.MustNewVaryingDataType(V1NeighbourMessage{})
+
+	return VersionedNeighbourMessage(vdt)
+}
+
+// Set updates the current VDT value to be `val`
+func (vnm *VersionedNeighbourMessage) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*vnm)
+	err = vdt.Set(val)
+	if err != nil {
+		return
+	}
+	*vnm = VersionedNeighbourMessage(vdt)
+	return
+}
+
+// Value returns the current VDT value
+func (vnm *VersionedNeighbourMessage) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*vnm)
+	return vdt.Value()
+}
+
+// V1NeighbourMessage represents a network-level neighbour message
+type V1NeighbourMessage struct {
+	Round  uint64
+	SetID  uint64
```

Comment:
this is actually a struct containing a u64 in the substrate codebase [(line)](https://github.com/paritytech/substrate/blob/86a0fc25e3aaeee273b6b3706f2aef99df72a21e/core/finality-grandpa/src/communication/mod.rs#L633).  Maybe provide a comment here referencing this.  If they add an attribute to this struct it will break encoding/decoding.

Reply:
Any idea why they are using a struct? 🤔

Reply:
I would say that is a common practice in rust wrapper primitive types within another meaningful type, I believe its called [zero-cost abstraction](https://blog.polybdenum.com/2021/08/09/when-zero-cost-abstractions-aren-t-zero-cost.html)